src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -420,9 +421,17 @@ protected static boolean isAltAfterAssembly(final PileupElement element, final b
      * @param priorList priors within the current ActiveRegion
      * @return prior VCs representing the same variant position as call
      */
-    List<VariantContext> getMatchingPriors(final Locatable curPos, final VariantContext call, final List<VariantContext> priorList) {
+    private List<VariantContext> getMatchingPriors(final Locatable curPos, final VariantContext call, final List<VariantContext> priorList) {
         final int position = call != null ? call.getStart() : curPos.getStart();
-        return priorList.stream().filter(vc -> position == vc.getStart()).collect(Collectors.toList());
+        final List<VariantContext> matchedPriors = new ArrayList<>(priorList.size());
+        // NOTE: a for loop is used here because this method ends up being called per-pileup, per-read and thus using faster alternates saves runtime
```

Comment:
" thus using faster alternates saves runtime" -> "using a loop instead of streaming saves runtime"